src/gpu/intel/ocl/ocl_gpu_engine.cpp

```diff
@@ -249,6 +250,9 @@ status_t ocl_gpu_engine_t::build_program_from_source(
     std::string pp_code_str = pp_code.str();
     const char *pp_code_str_ptr = pp_code_str.c_str();
 
+    src = {pp_code_str};
+    if (src) { options += " -g -s " + std::string(src.name()); }
```

Comment:
It seems to be the only place where program_src_t is needed.
Why introduce program_src_t, and not do everything in this function directly?

Reply:
The goal here is that this file is temporary and is deleted after it is no longer necessary. In particular, this means that this file is necessary so long the resulting `compute::kernels` is instantiated, so that this file can be referenced by a debugger.  The `program_src_t` structure is largely responsible for handling this lifetime management. We will also need a similar file if we are to introduce debug information for our [ir based](https://github.com/oneapi-src/oneDNN/tree/main/src/gpu/intel/jit/ir) kernels as currently all nGEN debug information just maps to the code which [lowers the IR to nGEN](https://github.com/oneapi-src/oneDNN/blob/main/src/gpu/intel/jit/codegen/codegen.cpp).

Reply:
ok I see. Indeed we need to pass something around, thanks.